### DISCUSSION TITLE
Enabling RGBW addressable LEDs

### DISCUSSION THREAD

**Discussion by maskari** [arendst/Tasmota#10894] [Q&A] [answered]
Hi I am trying to use WS2812 RGBW addressable LEDs with Tasmota but have been unsuccessful so far. Using regular RGB leds works perfectly fine but the RGBW ones get color assignments going to the wrong LEDs. I checked the MQTT and console responses from the device (ESP8266) and tasmota seems to ignore any white values I put in and only accepts RGB values.

Is there a specific toggle I need to enable to get tasmota to use RGBW instead of RGB?

Thank you

**Comment by sfromis** [comment, marked as answer]
It may not be "classic" WS2812B you have, but some variant with a different color order. If you build your own Tasmota binary, you can select neopixel variant.
```
#define USE_WS2812                               // WS2812 Led string using library NeoPixelBus (+5k code, +1k mem, 232 iram) - Disable by //
//  #define USE_WS2812_DMA                         // DMA supports only GPIO03 (= Serial RXD) (+1k mem). When USE_WS2812_DMA is enabled expect Exceptions on Pow
//  #define USE_WS2812_INVERTED                    // Use inverted data signal
  #define USE_WS2812_HARDWARE  NEO_HW_WS2812     // Hardware type (NEO_HW_WS2812, NEO_HW_WS2812X, NEO_HW_WS2813, NEO_HW_SK6812, NEO_HW_LC8812, NEO_HW_APA106, NEO_HW_P9813)
  #define USE_WS2812_CTYPE     NEO_GRB           // Color type (NEO_RGB, NEO_GRB, NEO_BRG, NEO_RBG, NEO_RGBW, NEO_GRBW)
```

**Reply by maskari** [reply]
Thank you! I had to build my own binary and select NEO_GRBW to get it to work correctly.
